你好启动后报加载错误。ImportError: cannot import name 'BaseConfig' from 'nonebot.config' (D:\yunzai\nonebot\qq2\.venv\lib\site-packages\nonebot\config.py)

### YetAnotherPicSearch 、 PicImageSearch 、 go-cqhttp 、 nonebot 、 Python 版本及操作系统

。

### 列出安装的 Python 包

。

### 如何复现

安装这个插件然后报错

### 期望行为

正常启动

### 实际行为

02-15 15:19:07 [ERROR] nonebot | Failed to import "YetAnotherPicSearch"
Traceback (most recent call last):
  File "<string>", line 14, in <module>
  File "D:\yunzai\nonebot\qq2\.venv\lib\site-packages\nonebot\plugin\load.py", line 130, in load_from_toml
    return load_all_plugins(plugins, plugin_dirs)
  File "D:\yunzai\nonebot\qq2\.venv\lib\site-packages\nonebot\plugin\load.py", line 64, in load_all_plugins
    return manager.load_all_plugins()
  File "D:\yunzai\nonebot\qq2\.venv\lib\site-packages\nonebot\plugin\manager.py", line 176, in load_all_plugins
    return set(
  File "D:\yunzai\nonebot\qq2\.venv\lib\site-packages\nonebot\plugin\manager.py", line 177, in <genexpr>
    filter(None, (self.load_plugin(name) for name in self.available_plugins))
> File "D:\yunzai\nonebot\qq2\.venv\lib\site-packages\nonebot\plugin\manager.py", line 142, in load_plugin
    module = importlib.import_module(name)
  File "C:\Users\USER451533\AppData\Local\Programs\Python\Python310\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "D:\yunzai\nonebot\qq2\.venv\lib\site-packages\nonebot\plugin\manager.py", line 242, in exec_module
    super().exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "D:\yunzai\nonebot\qq2\.venv\lib\site-packages\YetAnotherPicSearch\__init__.py", line 23, in <module>
    from .ascii2d import ascii2d_search
  File "D:\yunzai\nonebot\qq2\.venv\lib\site-packages\YetAnotherPicSearch\ascii2d.py", line 7, in <module>
    from .config import config
  File "D:\yunzai\nonebot\qq2\.venv\lib\site-packages\YetAnotherPicSearch\config.py", line 4, in <module>
    from nonebot.config import BaseConfig
ImportError: cannot import name 'BaseConfig' from 'nonebot.config' (D:\yunzai\nonebot\qq2\.venv\lib\site-packages\nonebot\config.py)

--- a/README.md
+++ b/README.md
@@ -3,7 +3,7 @@
 # YetAnotherPicSearch
 
 [![license](https://img.shields.io/github/license/NekoAria/YetAnotherPicSearch)](https://raw.githubusercontent.com/NekoAria/YetAnotherPicSearch/master/LICENSE)
-![python](https://img.shields.io/badge/python-3.8+-blue)
+![python](https://img.shields.io/badge/python-3.9+-blue)
 [![release](https://img.shields.io/github/v/release/NekoAria/YetAnotherPicSearch)](https://github.com/NekoAria/YetAnotherPicSearch/releases)
 
 基于 [nonebot2](https://github.com/nonebot/nonebot2) 及 [PicImageSearch](https://github.com/kitUIN/PicImageSearch) 的另一个 Nonebot 搜图插件

--- a/YetAnotherPicSearch/config.py
+++ b/YetAnotherPicSearch/config.py
@@ -14,7 +14,6 @@ else:
         __field: str,
         *fields: str,
         mode: Literal["before", "after", "wrap", "plain"] = "after",
-        check_fields: Optional[bool] = None,
     ):
         return validator(__field, *fields, pre=(mode == "before"), allow_reuse=True)
 
@@ -63,6 +62,12 @@ class Config(BaseModel):
         "konachan.com",
         "pixiv.net",
     ]
+
+    @field_validator("saucenao_api_key", mode="before")
+    def saucenao_api_key_validator(cls, v: str) -> str:
+        if not v:
+            raise ValueError("请配置 saucenao_api_key 否则无法正常使用搜图功能！")
+        return v
 
     @field_validator("proxy", mode="before")
     def proxy_validator(cls, v: Optional[str]) -> Optional[str]:

--- a/YetAnotherPicSearch/config.py
+++ b/YetAnotherPicSearch/config.py
@@ -1,11 +1,28 @@
-from typing import List, Optional
+from typing import List, Literal, Optional, Set
 
-from nonebot import get_driver
-from nonebot.config import BaseConfig
-from pydantic import Extra, validator
+from nonebot import get_plugin_config
+from nonebot.compat import PYDANTIC_V2
+from pydantic import BaseModel
+
+if PYDANTIC_V2:
+    from pydantic import field_validator  # type: ignore
+
+else:
+    from pydantic import validator
+
+    def field_validator(
+        __field: str,
+        *fields: str,
+        mode: Literal["before", "after", "wrap", "plain"] = "after",
+        check_fields: Optional[bool] = None,
+    ):
+        return validator(__field, *fields, pre=(mode == "before"), allow_reuse=True)
 
 
-class Config(BaseConfig):
+class Config(BaseModel):
+    superusers: Set[str]
+    nickname: Set[str]
+
     # 触发搜图的关键词
     search_keyword: str = "搜图"
     # 只响应含有搜图关键词的消息 (优先级高于 search_immediately)
@@ -47,20 +64,11 @@ class Config(BaseConfig):
         "pixiv.net",
     ]
 
-    class Config:
-        extra = Extra.allow
-
-    @validator("saucenao_api_key", pre=True)
-    def saucenao_api_key_validator(cls, v: str) -> str:
-        if not v:
-            raise ValueError("请配置 saucenao_api_key ，否则无法正常使用搜图功能！")
-        return v
-
-    @validator("proxy", pre=True)
+    @field_validator("proxy", mode="before")
     def proxy_validator(cls, v: Optional[str]) -> Optional[str]:
-        if v and v.startswith("socks://"):
+        if isinstance(v, str) and v.startswith("socks://"):
             raise ValueError('请修改代理地址为 "socks5://" 或 "socks4://" 的格式，具体取决于你的代理协议！')
         return v
 
 
-config = Config(**get_driver().config.dict())
+config = get_plugin_config(Config)

--- a/YetAnotherPicSearch/utils.py
+++ b/YetAnotherPicSearch/utils.py
@@ -48,7 +48,11 @@ async def get_image_bytes_by_url(
 ) -> Optional[bytes]:
     _url = URL(url)
     referer = f"{_url.scheme}://{_url.host}/"
-    headers = {"Referer": referer, **DEFAULT_HEADERS}
+    headers = (
+        DEFAULT_HEADERS
+        if _url.host.endswith("qpic.cn")
+        else {"Referer": referer, **DEFAULT_HEADERS}
+    )
     async with AsyncClient(
         headers=headers,
         cookies=parse_cookies(cookies),

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -25,7 +25,7 @@ keywords = [
 ]
 
 [tool.poetry.dependencies]
-python = ">=3.8,<4"
+python = ">=3.9,<4"
 arrow = "^1.3.0"
 cachetools = "^5.3.2"
 httpx = "^0.25.2"
@@ -33,6 +33,7 @@ lxml = "^4.9.3"
 nonebot-adapter-onebot = "^2.4.1"
 nonebot2 = "^2.2.0"
 PicImageSearch = "^3.9.7"
+pydantic = "^2.6.4"
 pyquery = "^2.0.0"
 python-socks = { extras = ["asyncio"], version = "^2.4.4", optional = true }
 socksio = { version = "^1.0.0", optional = true }
@@ -43,6 +44,7 @@ tenacity = "^8.2.3"
 socks = ["python-socks", "socksio"]
 
 [tool.poetry.dev-dependencies]
+pre-commit = "^3.6.2"
 
 [tool.mypy]
 python_version = "3.8"

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -5,25 +5,36 @@ description = "Yet Another Picture Search Nonebot Plugin"
 authors = ["NekoAria"]
 license = "GPL-3.0-only"
 readme = "README.md"
-packages = [
-    { include = "YetAnotherPicSearch" },
-]
+packages = [{ include = "YetAnotherPicSearch" }]
 homepage = "https://github.com/NekoAria/YetAnotherPicSearch"
 repository = "https://github.com/NekoAria/YetAnotherPicSearch"
-keywords = ["nonebot", "ascii2d", "baidu", "e-hentai", "google", "iqdb", "saucenao", "tracemoe", "yandex", "anime", "danbooru", "doujin", "pixiv"]
+keywords = [
+    "nonebot",
+    "ascii2d",
+    "baidu",
+    "e-hentai",
+    "google",
+    "iqdb",
+    "saucenao",
+    "tracemoe",
+    "yandex",
+    "anime",
+    "danbooru",
+    "doujin",
+    "pixiv",
+]
 
 [tool.poetry.dependencies]
-python = "^3.8"
+python = ">=3.8,<4"
 arrow = "^1.3.0"
 cachetools = "^5.3.2"
 httpx = "^0.25.2"
 lxml = "^4.9.3"
-nonebot-adapter-onebot = "^2.3.1"
-nonebot2 = "^2.1.2"
+nonebot-adapter-onebot = "^2.4.1"
+nonebot2 = "^2.2.0"
 PicImageSearch = "^3.9.7"
-pydantic = "^1.10.13"
 pyquery = "^2.0.0"
-python-socks = {extras = ["asyncio"], version = "^2.4.4", optional = true}
+python-socks = { extras = ["asyncio"], version = "^2.4.4", optional = true }
 socksio = { version = "^1.0.0", optional = true }
 shelved-cache = "^0.3.1"
 tenacity = "^8.2.3"
@@ -35,9 +46,7 @@ socks = ["python-socks", "socksio"]
 
 [tool.mypy]
 python_version = "3.8"
-plugins = [
-    "pydantic.mypy"
-]
+plugins = ["pydantic.mypy"]
 ignore_missing_imports = true
 implicit_reexport = true
 pretty = true

